pkg/recipes/terraform/config/providers/azure.go

```diff
@@ -119,30 +120,62 @@ func fetchAzureCredentials(ctx context.Context, azureCredentialsProvider credent
 	credentials, err := azureCredentialsProvider.Fetch(ctx, credentials.AzureCloud, "default")
 	if err != nil {
 		if errors.Is(err, &secret.ErrNotFound{}) {
-			logger.Info("AWS credentials are not registered, skipping credentials configuration.")
+			logger.Info("Azure credentials are not registered, skipping credentials configuration.")
 			return nil, nil
 		}
 
 		return nil, err
 	}
 
-	if credentials == nil || credentials.ClientID == "" || credentials.TenantID == "" || credentials.ClientSecret == "" {
+	switch credentials.Kind {
```

Comment:
@kachawla please review this file. not sure if I did this correctly.

Reply:
sure, didn't get a chance to finish looking at everything, will look into this file today